pkg/pillar/devicenetwork/pbr.go

```diff
@@ -88,12 +88,97 @@ func makeNetlinkRule(ifindex int, p net.IPNet, addForSubnet bool) *netlink.Rule
 	return r
 }
 
+func makeNetlinkGatewayRule(subnet net.IPNet, gateway net.IP, prio int) *netlink.Rule {
+	r := netlink.NewRule()
+	r.Table = syscall.RT_TABLE_LOCAL
+	r.Priority = prio
+	r.Family = HostFamily(gateway)
+
+	r.Src = &subnet
+	var g net.IPNet
+	g = HostSubnet(gateway)
+	r.Dst = &g
+
+	return r
+}
+
+func makeInwardNetlinkRule(ifindex int, p net.IPNet, addForSubnet bool, prio int) *netlink.Rule {
+	r := netlink.NewRule()
+	r.Table = baseTableIndex + ifindex
+	r.Priority = prio
+	r.Family = HostFamily(p.IP)
+
+	var subnet net.IPNet
+	if addForSubnet {
+		subnet = p
+	} else {
+		subnet = HostSubnet(p.IP)
+	}
+	r.Dst = &subnet
+
+	return r
+}
+
+// MoveDownLocalIPRule : Move IP rule that matches local destined packets below network instance rules.
+func MoveDownLocalIPRule(log *base.LogObject, prio int) {
+	// IPv4
+	r := netlink.NewRule()
+	r.Table = syscall.RT_TABLE_LOCAL
+	r.Priority = prio
+	r.Family = syscall.AF_INET
+	if err := netlink.RuleAdd(r); err != nil {
+		log.Errorf("MoveDownLocalIPRule: RuleAdd %v failed with %s", r, err)
+		return
+	}
+	r.Priority = 0
+	_ = netlink.RuleDel(r)
```

Comment:
It would be good to check and log errors here as well.

Reply:
will add error checking.